stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,653 @@
+    """
+    def set_timeout(self, timeout=None):
+        if isinstance(timeout, int):
+            self.timeout = timeout
+        else:
+            self.timeout = None
+
+    """
+        Returns the current state of the proxy
+        ** Affects: None **
+    """
+    def get_timeout(self):
+        return self.timeout
+
+    """
+        # Validate
+        @timeout
+    """
+    resp_validate_schematic = None
+    resp_validate_status_code = None
+    resp_validate_strict_type = None
+
+    def set_resp_validation(self, strict_type=None, schematic=None, status_code=None):
+        self.resp_validate_schematic = schematic
+        self.resp_validate_status_code = status_code
+        if strict_type in self.available_response_types:
+            self.resp_validate_strict_type = strict_type
+
+    """
+        Returns the current state of the validations
+        ** Affects: None **
+    """
+    def get_resp_validate_schematic(self):
+        return self.resp_validate_schematic
+
+    def get_resp_validate_status_code(self):
+        return self.resp_validate_status_code
+
+    def get_resp_validate_strict_type(self):
+        return self.resp_validate_strict_type
+
+    """
+        Used for unit testing only
+    """
+    def mount_adapter(self, adapter):
+        self.session.mount('mock://', adapter)
+
+    """
+        Uses the Retry object from requests allowing the user to apply the same kwargs parameters
+
+        @kwargs
+            List items accepted by the Retry function
+    """
+    def set_retry_policy(self, **kwargs):
+        adapter = None
+
+        if len(kwargs) > 0:
+            self.retry_strategy = Retry(**kwargs)
+            adapter = HTTPAdapter(max_retries=self.retry_strategy)
+
+        self.session.mount("https://", adapter)
+        self.session.mount("http://", adapter)
+
+    """
+        Returns the current state of the retry policy
+        ** Affects: None **
+    """
+    def get_retry_policy(self):
+        return self.retry_strategy
+
+    """
+        Set the current state of the SSL verification, Set to False so that SSL is not verified
+
+        @verify
+            True or False to enable or disable ssl verification
+    """
+    def set_ssl_verify(self, verify=None):
+        if isinstance(verify, bool):
+            self.session.verify = verify
+        else:
+            self.session.verify = True
+
+    """
+        Returns the current state of the SSL
+        ** Affects: None **
+    """
+    def get_ssl_verify(self):
+        return self.session.verify
+
+    """
+        This function is used to send out the request that has been build up
+        It has one of two return types
+            - The response from the request
+            - Or None if the response fails any validations unless no validation was defined and the response is JSON
+    """
+    def send(self):
+        # Send the request out
+        response = self.session.send(self.request_object.prepare(),
```

Comment:
It would be better separation of concerns to let the caller create the request object before and then hand that to this method as a parameter. Otherwise you get a method on a class that you can only call under certain circumstances and you now first have to figure out a bunch of prerequisites before you can use it. In general we should strive for as much referential transparency as possible.